src/meson.build

```diff
@@ -94,34 +94,13 @@ fwupdmgr = executable(
   install : true,
   install_dir : bindir
 )
-endif
 
-if get_option('agent')
-fwupdagent = executable(
-  'fwupdagent',
-  sources : [
-    'fu-agent.c',
-    'fu-security-attr.c',
-    'fu-util-common.c',
-    systemd_src,
-  ],
-  include_directories : [
-    root_incdir,
-    fwupd_incdir,
-    fwupdplugin_incdir,
-  ],
-  dependencies : [
-    libfwupd_deps,
-    libxmlb,
-    client_dep,
-  ],
-  link_with : [
-    fwupd,
-    fwupdplugin,
-  ],
-  install : true,
-  install_dir : bindir
-)
+# for compatibility
+meson.add_install_script(
+  'sh', '-c',
+  'ln -fs fwupdmgr @0@@1@'.format(
```

Comment:
did you try to just have it just store symlink in the tree and install the symlink instead of making it explicitly at install time?  I feel like this approach you did might be more fragile with packaging.

Reply:
I tried this; it gets horrible when trying to install into destdirs and things like that. See https://github.com/mesonbuild/meson/issues/1602